pl_bolts/models/autoencoders/basic_ae/basic_ae_module.py

```diff
@@ -42,25 +38,24 @@ def __init__(
         super().__init__()
         self.save_hyperparameters()
 
-        # link default data
-        if datamodule is None:
-            datamodule = MNISTDataModule(data_dir=self.hparams.data_dir, num_workers=self.hparams.num_workers)
-
-        self.datamodule = datamodule
-
-        self.img_dim = self.datamodule.size()
-
-        self.encoder = self.init_encoder(self.hparams.hidden_dim, self.hparams.latent_dim,
-                                         self.hparams.input_width, self.hparams.input_height)
+        self.encoder = self.init_encoder(
+            self.hparams.hidden_dim,
+            self.hparams.latent_dim,
+            self.hparams.input_channels,
+            self.hparams.input_width,
+            self.hparams.input_height,
+        )
         self.decoder = self.init_decoder(self.hparams.hidden_dim, self.hparams.latent_dim)
 
-    def init_encoder(self, hidden_dim, latent_dim, input_width, input_height):
-        encoder = AEEncoder(hidden_dim, latent_dim, input_width, input_height)
+    def init_encoder(self, hidden_dim, latent_dim, input_channels, input_height, input_width):
+        encoder = AEEncoder(hidden_dim, latent_dim, input_channels, input_height, input_width)
         return encoder
 
     def init_decoder(self, hidden_dim, latent_dim):
-        c, h, w = self.img_dim
-        decoder = Decoder(hidden_dim, latent_dim, w, h, c)
+        # c, h, w = self.img_dim
+        decoder = Decoder(
+            hidden_dim, latent_dim, self.hparams.input_width, self.hparams.input_height, self.hparams.input_channels
```

Comment:
let's keep this order if it is everywhere else

Reply:
it should be c, h, w right? I started switching over to that from 'c, w, h', which is wrong, if I'm not mistaken? (that's what I kept seeing in this file, but I believe its mixed up)

Reply:
just see that the order is different in several places...